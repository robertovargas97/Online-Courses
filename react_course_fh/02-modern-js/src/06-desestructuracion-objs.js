// Asignacion Desestructurante 
// https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

const hero = { name: "Tony", age: 45, key: "Ironman" };

const returnPerson = ({ name: heroName, age: heroAge, key: heroKey, range = null }) => {
    const result = {
        keyName: heroKey,
        age: heroAge,
        latLng: {
            lat: 1234.3455,
            lng: 3443.333
        }
    }
    return result
}


const { keyName, age, latLng: { lat, lng } } = returnPerson(hero);
console.log(keyName, age);
console.log(lat, lng);
