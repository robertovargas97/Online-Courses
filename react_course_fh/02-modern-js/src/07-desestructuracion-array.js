// Asignacion Desestructurante 
// https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

const characters = ["Goku", "Vegeta", "Trunks"];

const [goku, , trunks] = characters;


const returnArreglo = () => {
    return ["ABC", 123];
}

const [letters] = returnArreglo();

console.log(letters);


const use_State = (value) => {
    return [value, () => { console.log("Hello") }];
}

const [name, setName] = use_State("Goku");

console.log(name);
setName();
