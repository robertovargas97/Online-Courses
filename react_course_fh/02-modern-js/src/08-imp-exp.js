// Default export is withuout the brackets
// import heroes, { owners } from "./data/heroes";
import heroes from "./data/heroes";

export const getHeroById = (id) => {
    return heroes.find((hero) => hero.id === id);
}

export const getHeroByOwner = (owner) => {
    return heroes.filter((hero) => hero.owner === owner);
}


// console.log(getHeroById(2));
// console.log(getHeroByOwner("DC"));
// console.log(heroes);