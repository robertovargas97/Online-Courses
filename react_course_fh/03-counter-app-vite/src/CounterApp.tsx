import { useState } from 'react';



interface counterAppPropTypes {
    value: number;
}

// type clickEvent = React.MouseEvent<HTMLButtonElement, MouseEvent>;



export const CounterApp: React.FC<counterAppPropTypes> = ({ value }): JSX.Element => {

    const [counter, setCounter] = useState(value);

    const increaseValue = (): void => {
        setCounter(counter + 1);
        // setCounter((c) => c + 1)
    };

    const decreaseValue = (): void => {
        setCounter(counter - 1);
    };

    const resetValue = (): void => {
        setCounter(value);
    };

    const component: JSX.Element =
        <>
            <h1>Counter App</h1>
            <h2 data-testid="initial-value"> {counter} </h2>

            <button data-testid="button-increase" onClick={increaseValue}> +1 </button>
            <button data-testid="button-decrease" onClick={decreaseValue}> -1 </button>
            <button data-testid="button-reset" onClick={resetValue}> Reset </button>
        </>


    return component
}
