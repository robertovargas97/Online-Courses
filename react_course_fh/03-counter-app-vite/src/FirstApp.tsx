
// It is recommended to define things that will not change, out of the Functional Component
// When this is out of the component, React will not re-render it
type myObject = { name: string, pet: boolean };
const myObject: myObject = { name: "Robertyo", pet: true };

const returnMessage = (): string => {
    const newMessage: string = "My first App";
    return newMessage
}

interface propTypes {
    subTitle?: number
    title: string;
}

// Component to render
export const FirstApp: React.FC<propTypes> = ({ title, subTitle }): JSX.Element => {

    const component: JSX.Element =
        // <> : Represents a fragment
        <>
            <h1 data-testid="test-title">
                {title}
            </h1>
            <br></br>
            <h2 data-testid="test-subTitle">
                {subTitle}
            </h2>

            <br></br>
            {returnMessage()}

            <code>
                {JSON.stringify(myObject)}
            </code>

            <p>This is a text</p>
        </>

    return (component)
}
