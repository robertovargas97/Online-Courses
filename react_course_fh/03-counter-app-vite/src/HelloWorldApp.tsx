// Component to render
export const HelloWorldApp = (): JSX.Element => {
    return (
        <div> Hello World App </div>
    );
}
