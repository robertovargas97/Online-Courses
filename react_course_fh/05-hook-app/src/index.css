body {
  padding: 20px;
  background-color: rgb(28, 39, 42);
  color: white;
}

button {
  margin-right: 10px;
}