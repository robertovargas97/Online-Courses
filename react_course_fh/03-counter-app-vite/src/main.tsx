import React from 'react';
import ReactDOM from 'react-dom/client';

// import { HelloWorldApp } from './HelloWorldApp';
// import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';


// Import the styles globally
import "./styles.css"


// Root element
const root: HTMLElement = document.getElementById("root")!;

// Rendering the component
ReactDOM.createRoot(root).render(

    <React.StrictMode>
        {/* <HelloWorldApp /> */}

        {/* <FirstApp title='React Course' subTitle={123} /> */}

        <CounterApp value={1} />

    </React.StrictMode>

);
