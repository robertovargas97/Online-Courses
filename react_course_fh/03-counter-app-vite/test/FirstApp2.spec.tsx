import { render, screen } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp';


describe('<FirstApp> Test Cases Suit', () => {

    const title = "My Test Title";
    const subTitle = 1234;


    test('should match with the snapshot', () => {
        const { container } = render(<FirstApp title={title}></FirstApp>);
        // Allows to create a snapshot to make sure the component does not change
        expect(container).toMatchSnapshot();

    });

    test('should display the tittle "My Test Title"', () => {
        render(<FirstApp title={title}></FirstApp>);
        expect(screen.getByText(title)).toBeTruthy();
        // screen.debug();
    });

    test('should display the tittle "My Test Title" in an H1 tag', () => {
        render(<FirstApp title={title}></FirstApp>);
        const componentTitle = screen.getByRole('heading', { level: 1 }).innerHTML;
        expect(componentTitle).toContain(title);

    })

    test('should display the subTittle sent in the props', () => {
        render(
            <FirstApp title={title} subTitle={subTitle} />
        );

        const componentSubTitle = screen.getByRole('heading', { level: 2 }).innerHTML;

        // Ensure the text is present in the Component
        expect(componentSubTitle).toBe(subTitle.toString());

    });


});