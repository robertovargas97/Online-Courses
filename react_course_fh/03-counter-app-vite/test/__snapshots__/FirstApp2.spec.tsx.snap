// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstApp> Test Cases Suit should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    My Test Title
  </h1>
  <br />
  <h2
    data-testid="test-subTitle"
  />
  <br />
  My first App
  <code>
    {"name":"Robertyo","pet":true}
  </code>
  <p>
    This is a text
  </p>
</div>
`;
